### PR TITLE
[FIX] Save gifti with datatype None

### LINKED ISSUE #4855: [BUG] Some new test failures on s390x (big-endian) in 0.11.0
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Operating system

- [x] Linux
- [ ] Mac
- [ ] Windows

### Operating system version

Fedora Rawhide, or 41

### Python version

- [x] 3.13
- [ ] 3.12
- [ ] 3.11
- [ ] 3.10
- [ ] 3.9

### nilearn version

0.11.0

### Expected behavior

All tests pass

### Current behavior & error messages

On `s390x` (the sole big-endian primary architecture in Fedora) only:

```
FAILED nilearn/surface/tests/test_surface.py::test_load_save_data[foo.gii-expected_files0-unexpected_files0-True] - UnboundLocalError: cannot access local variable 'datatype' where it is not associated with a value
FAILED nilearn/surface/tests/test_surface.py::test_load_save_data[foo.gii-expected_files0-unexpected_files0-False] - UnboundLocalError: cannot access local variable 'datatype' where it is not associated with a value
FAILED nilearn/surface/tests/test_surface.py::test_load_save_data[foo_hemi-L_T1w.gii-expected_files1-unexpected_files1-True] - UnboundLocalError: cannot access local variable 'datatype' where it is not associated with a value
FAILED nilearn/surface/tests/test_surface.py::test_load_save_data[foo_hemi-L_T1w.gii-expected_files1-unexpected_files1-False] - UnboundLocalError: cannot access local variable 'datatype' where it is not associated with a value
FAILED nilearn/surface/tests/test_surface.py::test_load_save_data[foo_hemi-R_T1w.gii-expected_files2-unexpected_files2-True] - UnboundLocalError: cannot access local variable 'datatype' where it is not associated with a value
FAILED nilearn/surface/tests/test_surface.py::test_load_save_data[foo_hemi-R_T1w.gii-expected_files2-unexpected_files2-False] - UnboundLocalError: cannot access local variable 'datatype' where it is not associated with a value
= 6 failed, 3508 passed, 16 skipped, 182 deselected, 646 xfailed, 1031 warnings in 4285.00s (1:11:25) =
```

I’ve been having trouble attaching the full build log to this report, but I’ll at least paste the full output from the failing tests into a follow-up comment.

### Steps and code to reproduce bug

The original report comes from a Fedora RPM build on real `s390x` hardware in Fedora infrastructure. Unfortunately, it’s hard to say how best to reproduce this outside of Fedora: a chroot with `qemu-user-static` or an emulated VM should work, but there are no `numpy` binary wheels for `s390x` on PyPI and building it is tedious. I know that’s not ideal. You might have better luck running the `s390x` version of Fedora in a (qemu-emulated) VM and installing the system versions of `nilearn`’s dependencies, or I’m happy to run test builds with any candidate patches on real hardware (to which I have *non-interactive* access for testing and building Fedora packages).

### PR DESCRIPTION
**Pull request by man-shu**
- Closes #4855 

Changes proposed in this pull request:

- use datatype None if data format is not recognized

### CODE REVIEW

**Comment by musicinmybrain** [comment]
I’m testing this on s390x. I should be able to report results tomorrow morning.

**Comment by musicinmybrain** [comment]
I tried this as a patch to 0.11.0, and it does not seem to have made a difference in the failing tests on `s390x`.

**Comment by musicinmybrain** [comment]
> I tried this as a patch to 0.11.0, and it does not seem to have made a difference in the failing tests on `s390x`.

Stop the presses! I’m mistaken. The patch wasn’t actually applied. I need to try it again.

**Review by bthirion — COMMENTED** [review]
A test is needed on this PR.

**Comment by musicinmybrain** [comment]
This *does* work in my testing! All of the tests that regressed in https://github.com/nilearn/nilearn/issues/4855 pass when I properly apply this PR as a patch for 0.11.0.

File: nilearn/surface/surface.py [hunk 1]
```diff
@@ -1558,7 +1558,8 @@ def _data_to_gifti(data, gifti_file):
         datatype = "NIFTI_TYPE_INT32"
     elif data.dtype == np.float32:
         datatype = "NIFTI_TYPE_FLOAT32"
-
+    else:
+        datatype = None
```
Comment (Remi-Gau) [inline nilearn/surface/surface.py:1562]: make sure we cover this in our tests 😉

Comment (man-shu) [reply nilearn/surface/surface.py:1562]: Done!

**Review by bthirion — APPROVED** [review]
LGTM.

**Review by Remi-Gau — APPROVED** [review]

**Pull request merged**